clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3125,6 +3140,13 @@ static void RenderFloatingPointOptions(const ToolChain &TC, const Driver &D,
       TrappingMathPresent = true;
       FPExceptionBehavior = "strict";
       break;
+    case options::OPT_fveclib:
+      VecLibArg = A;
+      if (llvm::is_contained(VecLibImpliesNoMathErrno, A->getValue())) {
+        MathErrno = false;
+        NoMathErrnoWasImpliedByVecLib = true;
+      }
```

Comment:
It would be unusual but I suggest 
```
NoMathErrnoWasImpliedByVecLib = llvm::is_contained(VecLibImpliesNoMathErrno, A->getValue());
if (NoMathErrnoWasImpliedByVecLib)
  MathErrno = false;
```
to cover the `-fveclib=ArmPL -fmath-errno -fveclib=LIBMVEC` scenario.

Reply:
Done (and added test cases for the two scenarios discussed in these comments) :+1: